rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/NodesRecoveryManager.java

```diff
@@ -75,8 +98,190 @@ public NodesLockRestorationManager apply(RMCore rmCore) {
         };
     }
 
-    public void restoreLocks(RMNode rmNode, Client caller) {
-        nodesLockRestorationManager.handle(rmNode, caller);
+    public void recoverNodeSourcesAndNodes() {
+        Collection<NodeSourceData> nodeSources = rmCore.getDbManager().getNodeSources();
+        logPersistedNodeSourceInfo(nodeSources);
+
+        for (NodeSourceData nodeSourceData : nodeSources) {
+            String nodeSourceName = nodeSourceData.getName();
+            if (NodeSource.DEFAULT_LOCAL_NODES_NODE_SOURCE_NAME.equals(nodeSourceName)) {
+                // will be recreated by SchedulerStarter
+                rmCore.getDbManager().removeNodeSource(nodeSourceName);
+            } else {
+                recoverDeployedNodeSourceIfNeeded(nodeSourceData, nodeSourceName);
+            }
+        }
+    }
+
+    public boolean recoverInfrastructureVariablesIfNeeded(String nodeSourceName, NodeSource nodeSourceToDeploy,
```

Comment:
"recoverInfrastructureVariablesIfNeeded" suggest that it recovers the infrastructure.

But this line ` rmCore.getDbManager().removeAllNodesFromNodeSource(nodeSourceName);` deletes all nodes it seems. 
I think a higher level should decide if the nodes need to be deleted if the infrastructure cannot be recovered.

Reply:
It is difficult to move it up because the decision can only be made after the infrastructure recovery.
I tried to improve anyway by renaming the method and the returned variable to be more precise.